src/coreclr/src/vm/appdomain.cpp

```diff
@@ -4580,6 +4580,8 @@ IUnknown *AppDomain::CreateBinderContext()
 
         // Initialize the assembly binder for the default context loads for CoreCLR.
         IfFailThrow(CCoreCLRBinderHelper::DefaultBinderSetupContext(DefaultADID, &m_pTPABinderContext));
+
+        SystemDomain::SystemFile()->SetAssemblyLoadContext(m_pTPABinderContext);
```

Comment:
Why do we need to set it explicitly? I would expect the constructor to set it for corelib too.

Reply:
I've mentioned it in my comment above. the PEFile (PEAssembly) for it is created during the ICLRRuntimeHost4::Start, but the related ALC much later during ICLRRuntimeHost4::CreateAppDomainWithManager.

Reply:
Maybe useful to add comment.